ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,188 @@
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+        assert response.context['sup_pro_id'] == supplier_product0.supplier_product_id
+        delivery_location_by_objects_filter = DeliveryLocation.objects.filter(
+                                                          user_name=supplier_product0.user_name,
+                                                          location=saved_client0.area)
+        assert delivery_location0.location != saved_client0.area
+        assert list(delivery_location_by_objects_filter) == []
+        delivery_location_by_the_web_view = response.context['deliveries']
+        assert list(delivery_location_by_the_web_view) == list(delivery_location_by_objects_filter)
+
+    @pytest.mark.django_db()
+    def test_invalid_find_delivery(self, client, invalid_supplier_product_id, client0_login):
+        with pytest.raises(ObjectDoesNotExist):
+            SupplierProduct.objects.get(supplier_product_id=invalid_supplier_product_id)
+        response = client.get('/ordered_product/new_order/find_delivery/'+invalid_supplier_product_id)
+        assert response.status_code == 302
+
+    @pytest.mark.django_db()
+    def test_valid_order_summary(self, client, supplier_product0, supplier_product0_id, delivery_location0,
+                                 client0_login):
+        supplier_product0.save()
+        delivery_location0.save()
+        supplier_product_by_objects_filter = SupplierProduct.objects.get(supplier_product_id=supplier_product0_id)
+        delivery_location_by_objects_filter = DeliveryLocation.objects.get(id=delivery_location0.id)
+        assert supplier_product0 == supplier_product_by_objects_filter
+        assert delivery_location0 == delivery_location_by_objects_filter
+        url = '/ordered_product/new_order/order_summary/'+str(supplier_product0_id)+','+str(delivery_location0.id)
+        response = client.get(url)
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+        supplier_products_by_the_web_view = response.context['supplier_product']
+        delivery_location_by_the_web_view = response.context['delivery_location']
+        assert supplier_products_by_the_web_view == supplier_product_by_objects_filter
+        assert delivery_location_by_the_web_view == delivery_location_by_objects_filter
+
+    @pytest.mark.django_db()
+    def test_invalid_order_summary(self, client, invalid_delivery_location_id, invalid_supplier_product_id,
+                                   client0_login):
+        with pytest.raises(ObjectDoesNotExist):
+            SupplierProduct.objects.get(supplier_product_id=invalid_supplier_product_id)
+            DeliveryLocation.objects.get(id=invalid_delivery_location_id)
```

Comment:
Which line is going to raise? Leave only that line inside the context manager.

Reply:
If it's both, then you need two such context managers as once the code raises an exception, the rest of the lines inside the context manager will not be executed.